src/sidebar/components/sidebar-content.js

```diff
@@ -1,202 +1,165 @@
-    if (
-      !store.frames().some(function (frame) {
-        return frame.uri;
-      })
-    ) {
-      // The document's URL isn't known so the document must still be loading.
-      return true;
-    }
-    return store.isFetchingAnnotations();
-  };
-
-  $scope.$on('sidebarOpened', function () {
-    analytics.track(analytics.events.SIDEBAR_OPENED);
-
-    streamer.connect();
-  });
-
-  this.$onInit = () => {
-    // If the user is logged in, we connect nevertheless
-    if (this.auth.status === 'logged-in') {
-      streamer.connect();
-    }
-  };
+  onLogin,
+  onSignUp,
+  loadAnnotationsService,
+  rootThread: rootThreadService,
+  streamer,
+}) {
+  const rootThread = useStore(store =>
+    rootThreadService.thread(store.getState())
+  );
 
-  $scope.$on(events.USER_CHANGED, function () {
-    streamer.reconnect();
+  // store state values
+  const focusedGroupId = useStore(store => store.focusedGroupId());
+  const hasAppliedFilter = useStore(store => store.hasAppliedFilter());
+  const isFocusedMode = useStore(store => store.focusModeEnabled());
+  const annotationsLoading = useStore(store => {
+    return !store.hasFetchedAnnotations() || store.isFetchingAnnotations();
   });
+  const isLoggedIn = useStore(store => store.isLoggedIn());
+  const linkedAnnotationId = useStore(store =>
+    store.directLinkedAnnotationId()
+  );
+  const linkedAnnotation = useStore(store =>
+    store.findAnnotationByID(linkedAnnotationId)
+  );
+  const directLinkedTab = linkedAnnotation
+    ? tabForAnnotation(linkedAnnotation)
+    : null;
+  const searchUris = useStore(store => store.searchUris());
+  const sidebarHasOpened = useStore(store => store.hasSidebarOpened());
+  const userId = useStore(store => store.profile().userid);
+
+  // The local `$tag` of a direct-linked annotation; populated once it
+  // has anchored: meaning that it's ready to be focused and scrolled to
+  const linkedAnnotationAnchorTag =
+    linkedAnnotation && linkedAnnotation.$orphan === false
+      ? linkedAnnotation.$tag
+      : null;
+
+  // Actions
+  const clearSelectedAnnotations = useStore(
+    store => store.clearSelectedAnnotations
+  );
+  const selectTab = useStore(store => store.selectTab);
 
-  $scope.$on(events.ANNOTATIONS_SYNCED, function (event, tags) {
-    // When a direct-linked annotation is successfully anchored in the page,
-    // focus and scroll to it
-    const selectedAnnot = firstSelectedAnnotation();
-    if (!selectedAnnot) {
-      return;
-    }
-    const matchesSelection = tags.some(function (tag) {
-      return tag === selectedAnnot.$tag;
-    });
-    if (!matchesSelection) {
-      return;
+  const hasDirectLinkedAnnotationError = useStore(store => {
+    if (annotationsLoading || !linkedAnnotationId) {
+      return false;
     }
-    focusAnnotation(selectedAnnot);
-    scrollToAnnotation(selectedAnnot);
-
-    store.selectTab(tabs.tabForAnnotation(selectedAnnot));
+    return !store.annotationExists(linkedAnnotationId);
```

Comment:
Since we're already looking up the linked annotation above to get `linkedAnnotation`, we don't need to look it up a second time here using `annotationExists`. This could be simplified to something like:

```js
const hasDirectLinkedAnnotationError = !annotationsLoading && linkedAnnotationId && !linkedAnnotation
````

(nb. I'm not sure what the exact falsey value of `linkedAnnotationId` is here)